pkg/controller/mastership/watcher.go

```diff
@@ -46,7 +46,13 @@ func (w *TopoWatcher) Start(ch chan<- controller.ID) error {
 			log.Debugf("Received topo event '%s'", event.Object.ID)
 			if relation, ok := event.Object.Obj.(*topoapi.Object_Relation); ok &&
 				relation.Relation.KindID == topoapi.CONTROLS {
-				ch <- controller.NewID(relation.Relation.TgtEntityID)
+				srcEntity, err := w.topo.Get(ctx, relation.Relation.SrcEntityID)
+				if err == nil {
+					if srcEntity.GetEntity().KindID == topoapi.ONOS_CONFIG {
```

Comment:
These two conditions can be done in one line

Reply:
Actually, we should probably check if err != nil and log the error, otherwise check the entity type... an if-elseif statement.